documentation/design/user-stories/shared-infra.adoc

```diff
@@ -0,0 +1,277 @@
+  authenticationServiceRef:
+    name: corp-auth
+    namespace: auth-space
+```
+
+Then apply this resource using the command line tools or the console.
+
+NOTE: Only a tenant with the name 'default' is allowed in a namespace.
+
+=== Disable messaging for a namespace by deleting a messaging tenant
+
+Simply delete the resource:
+
+```
+kubectl delete messagingtenant default -n myapp
+```
+
+
+=== Create a messaging tenant without specifying a plan
+
+
+Messaging tenant creates the following resource with the desired settings. The namespaces may be omitted,
+in which case the operator namespace is assumed.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingTenant
+metadata:
+  name: default
+  namespace: myapp
+spec:
+  messagingInfraRef:
+    name: myinfra
+    namespace: infra-space
+  authenticationServiceRef:
+    name: corp-auth
+    namespace: auth-space
+```
+
+=== Create an address without specifying a plan
+
+Messaging tenant creates the following resource with the desired settings. The namespaces may be omitted,
+in which case the operator namespace is assumed.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: myqueue
+  namespace: myapp
+spec:
+  queue: {}
+```
+
+=== Ability to configure DLQ, expiry, forwarders etc. per address
+
+Define a special system queue that gets created on all brokers:
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: mydlq
+  namespace: myapp
+spec:
+  systemQueue: {}
+---
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: myexpiry
+   namespace: myapp
+spec:
+  systemQueue: {}
+```
+
+Then create
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddress
+metadata:
+  name: myqueue
+  namespace: myapp
+spec:
+  queue:
+    # Note: a queue 'mydlq' must exist!
+    DeadLetterQueue: mydlq
+    # Note: a queue 'myexpiry' must exist!
+    ExpiryQueue: myexpiry
+
+    # Forwarders require a MessagingConnector to exist, in this case one that is named 'remote1'.
+    forwarders:
+    - name: fwd1
+      remoteAddress: remote1/queue1
+      direction: in
+```
+
+NOTE: the properties that are only applicable to a few types, are specified under the type object as shown above.
+
+=== CRUDLW a messaging endpoint in a namespace where messaging is enabled
+
+Create a `MessagingEndpoint` resource containing either `cluster`, `route`, `ingress`, `loadbalancer` or `nodeport` objects
+that specify the properties. The protocols supported are AMQP, AMQP+TLS,
+AMQP-WS and AMQP-WS+TLS.
+
+The default protocols depends on the endpoint type. For "Ingress" and "Route", the default
+protocols is AMQPS. For NodePort, LoadBalancer and Cluster, the default protocols are AMQP, AMQPS.
+
+==== Configure an in-cluster endpoint with TLS
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingEndpoint
+metadata:
+  name: endpoint1
+  namespace: myapp
+spec:
+  # TLS configuration for endpoint
+  tls: {}
+
+  protocols:
```

Comment:
I think in the status section, some part of the object needs to map from protocols to port numbers.

Reply:
Updated proposal with a sub-header on endpoint status section.